### PR TITLE
fixup strtok usage and handling of wrong parameters

### PR DESCRIPTION
**Pull request by BytesGalore**

### CODE REVIEW

File: sixlowpan/main.c [hunk 1]
```diff
@@ -34,17 +44,20 @@ void init(char *str)
...
-        printf("\tradio_address must be an 8 bit integer\n");
+
+    if (str_len < 6) {
+        printUsage();
+        return;
     }
 
+    command = strtok(NULL, " ");
+    r_addr = (uint16_t) strtol(strtok(NULL, " "), NULL, 10);
+
```
Comment (LudwigKnuepfer) [inline sixlowpan/main.c]: trailing whitespace

File: sixlowpan/main.c [hunk 2]
```diff
@@ -34,17 +44,20 @@ void init(char *str)
...
+        return;
     }
 
+    command = strtok(NULL, " ");
+    r_addr = (uint16_t) strtol(strtok(NULL, " "), NULL, 10);
+    
+    if ( r_addr == 0 ) {
+        printUsage();
+        return;
+    }
```
Comment (LudwigKnuepfer) [inline sixlowpan/main.c]: trailing whitespace

File: sixlowpan/main.c [hunk 3]
```diff
@@ -34,17 +44,20 @@ void init(char *str)
...
-
-    if (str_len < 5) {
-        printf("Usage: init {h | r | a | e} radio_address\n");
-        printf("\th\tinitialize as host\n");
-        printf("\tr\tinitialize as router\n");
-        printf("\ta\tinitialize as ad-hoc router\n");
-        printf("\tb\tinitialize as border router\n\n");
-        printf("\tradio_address must be an 8 bit integer\n");
+
+    if (str_len < 6) {
```
Comment (mehlis) [inline sixlowpan/main.c]: a string len of at least 6: the reason for this is that "init X" has at least 6 chars?
I know that this apporach is common in riot but I would be happy to see this approach:

calling strtok once you get the command name, no big deal. calling it again you get a char or null (for a parameter of init). you can check for null of the result of the second strtok call.

**Comment by LudwigKnuepfer** [comment]
Currently this project is broken, so it would be nice to merge this and postpone the cosmetics.

**Comment by BytesGalore** [comment]
I've hopefully cut off the trailing whitespaces and added a ckeck for the first string token representing the `init` command. I left the `str_len < 6` check to jump out if its impossible that the command is right. This avoids an unnecessary `strcmp`

**Comment by LudwigKnuepfer** [comment]
I've opened https://github.com/BytesGalore/projects/pull/1 instead of commenting.

**Comment by LudwigKnuepfer** [comment]
ACK

**Comment by LudwigKnuepfer** [comment]
@BytesGalore / user01: Could you please amend your authorship or the whole commit?

**Comment by BytesGalore** [comment]
Ok, what do I have to do? (I am also user01, the name has been mistakenly used from a user account on a VM)

**Comment by LudwigKnuepfer** [comment]
git rebase -i master has on line help... f to merge it into the previous commit, otherwise `git commit --amend --reset-author` to change the last commits author.

**Comment by LudwigKnuepfer** [comment]
ACK for the PR

I could not get the project it to actually do anything useful in native though. That is no reason to not merge this though. A failing test is a good thing (compared to no test) ;-)

I'll open an issue with regard to this.

**Comment by LudwigKnuepfer** [comment]
'sup?

**Pull request merged**